src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -229,22 +254,51 @@ private static void updateAuthorList(RepoConfiguration config) throws NoAuthorsW
         }
     }
 
-    private static void generateIndividualRepoReport(
-            CommitContributionSummary commitSummary, AuthorshipSummary authorshipSummary, String repoReportDirectory) {
-        CommitReportJson commitReportJson = new CommitReportJson(commitSummary, authorshipSummary);
-        FileUtil.writeJsonFile(commitReportJson, getIndividualCommitsPath(repoReportDirectory));
-        FileUtil.writeJsonFile(authorshipSummary.getFileResults(), getIndividualAuthorshipPath(repoReportDirectory));
+    /**
+     * Adds {@code failedRepoConfig} into the collated list of failed repos and logs into the list of errors in the
+     *  summary.
+     */
+    private static void handleBranchingFailed(RepoConfiguration failedRepoConfig) {
+        ErrorSummary.getInstance().addErrorMessage(failedRepoConfig.getDisplayName(),
+                String.format(LOG_BRANCH_DOES_NOT_EXIST, failedRepoConfig.getBranch()));
+        failedRepoConfigsList.add(failedRepoConfig);
+    }
+
+    /**
+     * Adds all {@code failedRepoConfigs} into the list of failed repos and logs into the list of errors in the
+     *  summary.
+     */
+    private static void handleCloningFailed(List<RepoConfiguration> failedRepoConfigs) {
+        for (RepoConfiguration failedConfig : failedRepoConfigs) {
+            ErrorSummary.getInstance().addErrorMessage(failedConfig.getDisplayName(),
+                    String.format(LOG_ERROR_CLONING, failedConfig.getLocation()));
+            failedRepoConfigsList.add(failedConfig);
+        }
+    }
+
+    /**
+     * Remove configs from {@code repoConfigs} that failed to clone or branch out previously.
+     */
+    private static void removeFailedRepoConfigs(List<RepoConfiguration> repoConfigs) {
+        repoConfigs.removeAll(failedRepoConfigsList);
     }
 
     /**
-    * Generates a report at the {@code repoReportDirectory}.
-    */
+     * Generates a report at the {@code repoReportDirectory}.
+     */
     public static void generateEmptyRepoReport(String repoReportDirectory, String displayName) {
         CommitReportJson emptyCommitReportJson = new CommitReportJson(displayName);
         FileUtil.writeJsonFile(emptyCommitReportJson, getIndividualCommitsPath(repoReportDirectory));
         FileUtil.writeJsonFile(Collections.emptyList(), getIndividualAuthorshipPath(repoReportDirectory));
     }
 
+    private static void generateIndividualRepoReport(
```

Comment:
Why reorder this part?

Reply:
I reordered it so that the methods that generate the reports are all clustered below

Reply:
OK